qt/python/mantidqt/mantidqt/widgets/instrumentview/presenter.py

```diff
@@ -23,14 +24,20 @@ class InstrumentViewPresenter(ObservingPresenter):
     It has no model as its an old widget written in C++ with out MVP
     """
 
+    """
+    @param ws : The workspace object OR workspace name.
+    """
     def __init__(self, ws, parent=None, window_flags=Qt.Window, ads_observer=None, view: InstrumentView=None):
         super(InstrumentViewPresenter, self).__init__()
         self.ws_name = str(ws)
 
         self.container = view
         if not self.container:
+            workspace = AnalysisDataService.retrieve(self.ws_name)
+            workspace.readLock()
             self.container = InstrumentView(parent=parent, presenter=self,
```

Comment:
At some point it might be nice to make the workspace locks operate as a Python context manager but in the absence of that could we put this in a finally block just in case an exception is thrown out of the construction:
```python
           try:
                self.container = InstrumentView(parent=parent, presenter=self,
                                                                     name=self.ws_name, window_flags=window_flags)
           finally:
                workspace.unlock()
```

and the workspace then remains locked.

Reply:
Ah yes good point! Possibly add another test as well for if exception thrown the workspace is unlocked?

Reply:
I've added the finally block and written a test to verify that the workspace is unlocked if the instrument view throws an exception.